hasher-matcher-actioner/hmalib/common/configs/tx_collab_config.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+DynamoDB backed CollaborationConfig objects.
+
+This may become a CollaborationConfigStoreBase, but since none of the pytx
+interfaces require it, keeping it as just another class for now.
+"""
+
+from dataclasses import dataclass, asdict, fields
+import typing as t
+
+from threatexchange.exchanges.collab_config import CollaborationConfigBase
+from threatexchange.utils.dataclass_json import (
+    dataclass_dumps,
+    dataclass_loads,
+)
+
+from hmalib.common.mappings import full_class_name, import_class
+from hmalib.common.config import HMAConfig, create_config, update_config
+
+
+def _config_name(clazz: t.Type[CollaborationConfigBase], name: str):
+    """Use this to make the EditableCollaborationConfig's name unique."""
+    return f"{clazz.__name__}/{name}"
+
+
+@dataclass
+class EditableCollaborationConfig(HMAConfig):
+    """
+    Store collaboration configs as HMAConfigs. Attributes of collaboration config
+    are stored as serialized json.
+
+    Clients are not expected to work with this class directly. Use methods
+    exposed in this module to play with concrete CollaborationConfigBase
+    objects.
+    """
+
+    name: str
+
+    # A fully qualified class. eg.
+    # threatexchange.exchanges.impl.fb_threatexchange_api.FBThreatExchangeCollabConfig
+    collab_config_class: str
+
+    # Use these attributes to instantiate a collab_config
+    attributes_json_serialized: str
+
+
+def _build_collab_config(
+    collab_config_class: str, attributes_serialized: str
+) -> CollaborationConfigBase:
+    cls = import_class(collab_config_class)
+    return dataclass_loads(attributes_serialized, cls)
+
+
+def get_collab_config(
```

Comment:
(devil's advocate): Why not a method on EditableCollaborationConfig? Same with the others as well.